services/web/server/src/simcore_service_webserver/folders/_folders_db.py

```diff
@@ -0,0 +1,223 @@
+        list_query = base_query.order_by(asc(getattr(folders_v2.c, order_by.field)))
+    else:
+        list_query = base_query.order_by(desc(getattr(folders_v2.c, order_by.field)))
+    list_query = list_query.offset(offset).limit(limit)
+
+    async with get_database_engine(app).acquire() as conn:
+        count_result = await conn.execute(count_query)
+        total_count = await count_result.scalar()
+
+        result = await conn.execute(list_query)
+        rows = await result.fetchall() or []
+        results: list[FolderDB] = [parse_obj_as(FolderDB, row) for row in rows]
+        return cast(int, total_count), results
+
+
+async def get_folder_db(
+    app: web.Application,
+    folder_id: FolderID,
+    product_name: ProductName,
+) -> FolderDB:
+    query = (
+        select(*_SELECTION_ARGS)
+        .select_from(folders_v2)
+        .where(
+            (folders_v2.c.product_name == product_name)
+            & (folders_v2.c.folder_id == folder_id)
+        )
+    )
+
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(query)
+        row = await result.first()
+        if row is None:
+            raise FolderAccessForbiddenError(
+                reason=f"Folder {folder_id} does not exist.",
+            )
+        return parse_obj_as(FolderDB, row)
+
+
+async def get_folder_for_user_or_workspace(
+    app: web.Application,
+    folder_id: FolderID,
+    product_name: ProductName,
+    user_id: UserID | None,
+    workspace_id: WorkspaceID | None,
+) -> FolderDB:
+    assert not (
+        user_id is not None and workspace_id is not None
+    ), "Both user_id and workspace_id cannot be provided at the same time. Please provide only one."
+
+    query = (
+        select(*_SELECTION_ARGS)
+        .select_from(folders_v2)
+        .where(
+            (folders_v2.c.product_name == product_name)
+            & (folders_v2.c.folder_id == folder_id)
+        )
+    )
+
+    if user_id:
+        query = query.where(folders_v2.c.user_id == user_id)
+    else:
+        query = query.where(folders_v2.c.workspace_id == workspace_id)
+
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(query)
+        row = await result.first()
+        if row is None:
+            raise FolderAccessForbiddenError(
+                reason=f"User does not have access to the folder {folder_id}. Or folder does not exist.",
+            )
+        return parse_obj_as(FolderDB, row)
+
+
+async def update_folder(
+    app: web.Application,
+    folder_id: FolderID,
+    name: str,
+    parent_folder_id: FolderID | None,
+    product_name: ProductName,
+) -> FolderDB:
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(
+            folders_v2.update()
+            .values(
+                name=name,
+                parent_folder_id=parent_folder_id,
+                modified=func.now(),
```

Comment:
I think this is the default. Perhaps it does not happen because of what i mentioned on the migration?